cpp/tests/merge/merge-test.cu

```diff
@@ -8,6 +8,7 @@
 #include <rmm/thrust_rmm_allocator.h>
 #include <cudf/column/column_factories.hpp>
 #include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/table_utilities.hpp>
```

Comment:
Can this file be a .cpp file? Compiles faster if there is no device code and no thrust::device_vector.

Also, please add a copyright / license header. 

Also rename to merge_test (underscore, not dash).